### PR TITLE
FIX: correct labels processing

### PR DESCRIPTION
**Pull request by mrakitin**
## How Has This Been Tested?

That was hot-fixed at SMI.

### CODE REVIEW

File: bluesky/magics.py [hunk 1]
```diff
@@ -324,16 +324,20 @@ def get_labeled_devices(user_ns=None, maxdepth=6):
...
-                    for label in labels:
-                        obj_list[label].append((key, obj))
+            if hasattr(obj, '_ophyd_labels_'):
+                # don't inherit parent labels
+                labels = obj._ophyd_labels_
+                for label in labels:
+                    obj_list[label].append((key, obj))
+
+                if is_parent(obj):
+                    for c_key, v in get_labeled_devices(user_ns={k: getattr(obj, k)
```
Comment (danielballan) [inline bluesky/magics.py]: This is a neat trick.

**Comment by danielballan** [comment]
Would like to experiment with the usability a little more before merging. Trying to get this working locally against a spoofed IOC....

**Comment by danielballan** [comment]
OK, I think this is a decent second pass. If this behavior seems sensible to you all, let's merge it and get a bug-fix release out. Future iterations may follow as we try to use it more in the wild.

Example configuration:

```python
from ophyd import EpicsMotor, Device, Signal, Component
mtr = EpicsMotor('', name='mtr', labels={'motors'})
from bluesky.magics import BlueskyMagics
get_ipython().register_magics(BlueskyMagics)


class Thing(Device):
    a = Component(EpicsMotor, '')
    b = Component(EpicsMotor, '')


class ThingHaver(Device):
    tA = Component(Thing)
    tB = Component(Thing, labels={'favorites'})

th = ThingHaver(name='th', labels={'motors', 'things'})
s = Signal(name='s', labels={'motors', 'favorites', 'stuff'})
```

Output:

```
In [2]: %wa
motors
  Positioner                     Value       Low Limit   High Limit  Offset
  mtr                            0           0           0           0
  th_tA_a                        0           0           0           0
  th_tA_b                        0           0           0           0
  th_tB_a                        0           0           0           0
  th_tB_b                        0           0           0           0

  Local variable name                    Ophyd name (to be recorded as metadata)
  mtr                                    mtr
  th                                     th
  s                                      s

things
  Positioner                     Value       Low Limit   High Limit  Offset
  th_tA_a                        0           0           0           0
  th_tA_b                        0           0           0           0
  th_tB_a                        0           0           0           0
  th_tB_b                        0           0           0           0

  Local variable name                    Ophyd name (to be recorded as metadata)
  th                                     th

favorites
  Positioner                     Value       Low Limit   High Limit  Offset
  th_tB_a                        0           0           0           0
  th_tB_b                        0           0           0           0

  Local variable name                    Ophyd name (to be recorded as metadata)
  th.tB                                  th_tB
  s                                      s

stuff
  Local variable name                    Ophyd name (to be recorded as metadata)
  s                                      s
```

**Comment by mrakitin** [comment]
I like the updates.

**Comment by danielballan** [comment]
Closes #828 only 9 months late. Sorry for the long wait on that, @afluerasu and @ambarb.

**Comment by danielballan** [comment]
Self-merged with @tacaswell and @mrakitin's blessing via Slack. Moving on to tag a micro release....

**Pull request merged**